include/router.h

```diff
@@ -42,19 +42,19 @@ class RouterT
         Handler3& operator = (Handler3&&) = default;
         ~Handler3() = default;
 
-        Handler3(const Handler& pre, const Handler& peri, const Handler& post)
-            : pre(pre), peri(peri), post(post)
+        Handler3(const Handler& pre_action, const Handler& action, const Handler& post_action)
+            : pre_action(pre_action), action(action), post_action(post_action)
```

Comment:
`action(action)`
Maybe, better to rename it to `thread_action` or `worker_action` or something like this.